src/btree/bt_vrfy.c

```diff
@@ -78,6 +78,12 @@ __verify_config(WT_SESSION_IMPL *session, const char *cfg[], WT_VSTUFF *vs)
     WT_RET(__wt_config_gets(session, cfg, "dump_pages", &cval));
     vs->dump_pages = cval.val != 0;
 
+    WT_RET(__wt_config_gets(session, cfg, "read_corrupt", &cval));
+    if (cval.val)
+        F_SET(session, WT_SESSION_VERIFY_READ_CORRUPT);
```

Comment:
All uses of this new setting are in functions here that have access to the `vs` structure. There is no need for a session-wide flag. I suggest adding a boolean to the vs structure and setting it, similar to the `vs->dump_*` fields.